moveit_core/planning_scene/test/test_planning_scene.cpp

```diff
@@ -139,6 +155,20 @@ TEST(PlanningScene, MakeAttachedDiff)
   ps->checkCollision(req, res);
 }
 
+TEST(PlanningScene, isStateValid)
+{
+  srdf::ModelSharedPtr srdf_model(new srdf::Model());
+  urdf::ModelInterfaceSharedPtr urdf_model;
+  loadRobotModels(urdf_model, srdf_model);
+
+  planning_scene::PlanningScenePtr ps(new planning_scene::PlanningScene(urdf_model, srdf_model));
+  robot_state::RobotState current_state = ps->getCurrentState();
+  if (ps->isStateColliding(current_state, "left_arm"))
+  {
+    EXPECT_FALSE(ps->isStateValid(current_state, "left_arm"));
```

Comment:
This test isn't very useful, isn't it?
I think tests should verify the correct working of the API. In this particular example, you should set up some robot poses, that are known to collide/not collide and validate the result of `isStateColliding()`.
Or you should set up some constraints that are fulfilled or not and validate `isStateFeasible()`.

Reply:
@rhaschke Thank you for feedback. Work was done in the limited amount of time during World MoveIt Day 2017. Hopefully I will find some time to add suggested improvements before next WMD 2018 :)

Reply:
This seems useful enough to me... you use two different calls to verify expected behavior of invalid states.

Reply:
It's worth picking up these little bits. The more tests the better.

![itssomething](http://i2.kym-cdn.com/photos/images/newsfeed/000/114/139/tumblr_lgedv2Vtt21qf4x93o1_40020110725-22047-38imqt.jpg)